### ISSUE TITLE
Add target attribute to NavLink

### ISSUE THREAD

**Issue by DrGFreeman** [facultyai/dash-bootstrap-components#304]
The [NavLink](https://dash-bootstrap-components.opensource.faculty.ai/l/components/nav) object currently does not have a `target` attribute.

Having a `target` attribute would be useful to open links in different browser tab using `target="_blank"`.

It is possible to work around this using `html.A` which has the `target` attribute:

```python
dbc.NavItem(
    html.A(
        "Link Text",
        href="link_url",
        target="_blank",
        className="nav-link"
    )
)
```

However, having the `target` attribute in `dbc.NavLink` would make the `NavLink` API more complete and avoid users switching `dbc.NavLink` and `html.A` depending on whether they want the link to open in a separate browser tab.
```python
dbc.NavItem(
    dbc.NavLink(
        "Link Text",
        href="link_url",
        target="_blank"
    )
)

Thanks!
```

**Comment by tcbegley** [comment]
Thanks for the suggestion @DrGFreeman, I'm happy to add this.

Having thought about it a bit, I think it only really makes sense to support this for external links (I can elaborate on that if you like), would that still give you the functionality you want or do you want to open another page of your app in a new tab?

**Comment by DrGFreeman** [comment]
@tcbegley, Thanks for the quick response!

Most cases I can think of are for external links. I could however see cases where a link would point to a static portion of the application, like an app's documentation for example, and this link would have to open in a new tab and leave the current tab unchanged.

I could also see another case where the link would open another independent app running on the same server but this should probably be considered as an external link.

I may be missing some background but the way I see it, the `NavLink` is a wrapper around `html.A` that applies the `"nav-link"` bootstrap class (at least, that's how it renders in html). From that perspective, it could implement the `target` attribute in the same manner that `html.A` does?

**Comment by tcbegley** [comment]
> I may be missing some background but the way I see it, the `NavLink` is a wrapper around `html.A` that applies the `nav-link` bootstrap class (at least, that's how it renders in html). From that perspective, it could implement the target attribute in the same manner that `html.A` does?

That's correct in the case that it is being used as an external link, and you are right that in that case it's a simple case of passing the target attribute to the `<a>` element.

However something a bit different happens when they are used as an internal link. Dash apps are [single page applications](https://en.wikipedia.org/wiki/Single-page_application). They give the illusion of allowing you to navigate between pages by modifying the contents of the address bar, and the page contents, but the page is never actually reloaded by the browser. 

The standard way of achieving this is to use `dcc.Link` and `dcc.Location` together with your own callbacks. You can see [here](https://github.com/plotly/dash-core-components/blob/dev/src/components/Link.react.js#L43-L53) what's happening when you click on a `dcc.Link`. It calls `preventDefault` (to stop the browser following the link as it normally would), then manually updates the address bar. This change would be noticed by `dcc.Location` and trigger a callback allowing you to update the page contents. `NavLink` behaves the same way when used for internal links, so although it renders as a `<a>`, the click behaviour is different to the default behaviour.

Anyway, that's a fairly long winded way of saying when you click on an internal `NavLink` the browser isn't doing anything, and in particular you aren't actually visiting a new page, so `target` has no effect.

Now, we could say that if `target` has been set, the link should be followed as a link by the browser. Note though that if you open the app in a new tab, the entire app will have to load again from scratch. If the initial load is slow it will be quite a different experience to following the link in the same tab.

Having thought about it a bit more, perhaps the solution is simply to say that `target` will only have an effect if the link is treated as an external link, which can already be controlled by the user by setting `external_link=True`. Then we don't have to do any inspection of the `target` property, or decide who should win if e.g. `target` has been set but `external_link=False` etc.

**Comment by DrGFreeman** [comment]
Thanks for explaining the difference between the `dcc.Link` and `html.A`!

> the solution is simply to say that `target` will only have an effect if the link is treated as an external link, which can already be controlled by the user by setting `external_link=True`.

Makes perfect sense!

**Comment by tcbegley** [comment]
Hey @DrGFreeman,

I've just made a prerelease that should resolve this issue. I've tested it at my end and all seems well, but if you get a chance to try it out and let me know if it works for you that will be much appreciated.

```
pip install dash-bootstrap-components==0.8.2rc1
```

I'll likely make a full release over the next few days.

**Comment by DrGFreeman** [comment]
Tested on a [simple demo app](https://github.com/DrGFreeman/YAMLDash/commit/29283ba65320eb1d69022fdc1db810af740c921e) and it works perfectly. Thanks a lot!

Note that I did not have to set the `external_link` attribute to `True` for the `target="_blank"` to work.

**Comment by tcbegley** [comment]
> Note that I did not have to set the external_link attribute to True for the target="_blank" to work.

That should only be necessary if you supply an internal / relative link (e.g. `/internal`) that you want to be treated like an external link. If you supply a link such as `https://github.com` it will automatically infer that it's an external link.

**Comment by tcbegley** [comment]
Resolved in `0.8.2`

**Comment by rafaelqntn** [comment]
Hello,
I don't think this is very important, but I got a couple of warnings from React side regarding the target attribute of a `<a>` tag with a boolean value (see the stack trace below).

The link I'm rendering is an internal link:
`
dbc.NavLink(id='logout', className='ml-auto', children="Logout", href="/")
`
From the debug I could get the values that are being sent to React to render the element:

![image](https://user-images.githubusercontent.com/10837826/76282803-11966580-6256-11ea-9d0e-2f9923974228.png)

The warning disappears when I set the link as `external_link=True` (which is not the case), but it gets the right string whatever I set in the python code as value to the target attribute (e.g. _self).

Reading this topic, I guess that the target attribute of internal links should not be included in the `rawPropos.` object, instead of having a `false` value.

Thank you!

I'm using the following component's versions:
`
dash==1.9.1
dash-bootstrap-components==0.8.3
dash-core-components==1.8.1
dash-html-components==1.0.2
dash-renderer==1.2.4
`

Warnings from React side:
`
Warning: Received `false` for a non-boolean attribute `target`.

If you want to write it to the DOM, pass a string instead: target="false" or target={value.toString()}.

If you used to conditionally omit it with target={condition && value}, pass target={condition ? value : undefined} instead.
    in a (created by t)
    in t (created by t)
    in t (created by CheckedComponent)
    in CheckedComponent (created by TreeContainer)
    in UnconnectedComponentErrorBoundary (created by Connect(UnconnectedComponentErrorBoundary))
    in Connect(UnconnectedComponentErrorBoundary) (created by TreeContainer)
    in TreeContainer (created by Connect(TreeContainer))
    in Connect(TreeContainer) (created by TreeContainer)
    in nav (created by hl)
    in hl (created by wl)
    in wl (created by CheckedComponent)
[...]
`

**Comment by tcbegley** [comment]
Hey @rafaelqntn 

Thanks for reporting this! I think it's good if dash-bootstrap-components isn't generating confusing errors and warnings in the javascript console for users that might be less familiar with React etc.

This should already be fixed thanks to #315 which hasn't made it into a release yet. It's available in the prerelease `0.8.4rc1` though, could you try installing that and see if you get the same error messages?

```
pip install dash-bootstrap-components==0.8.4rc1
```

**Comment by rafaelqntn** [comment]
Oh, okay! Thank you for your quick response.
Using the RC the console's output is clean once again.

I'm going to watch this repo and wait for the next release.

This lib is pretty amazing, I appreciate this!
